### DISCUSSION TITLE
Incrementally Calculated Positions error

### DISCUSSION THREAD

**Discussion by SheriffBuzz** [ImageMagick/ImageMagick#5894] [Help] [answered]
I am using version 7.1.0-51 Q16-HDRi x64 (windows)

I am trying to follow this example to get an incremental layer offset below
https://www.imagemagick.org/Usage/layers/#layer_calc_inc

![image](https://user-images.githubusercontent.com/83767022/208571045-66db3fec-331c-4b94-8266-daaee34ceb43.png)

I am getting an error when running the sample command. All my images are the same size.

here is the raw command I am running:

magick convert *.png -set page '+%[fx:u[t-1]page.x+u[t-1].w-8]+%[fx:u[t-1]page.y+4]' -background none -layers merge +repage out.png

I am getting the following error output:
convert: Expected operator at 'page.x+u[t-1].w-8' @ error/fx.c/GetOperator/2446.
convert: Expected operator at 'page.y+4' @ error/fx.c/GetOperator/2446.
convert: Expected operator at 'page.y+4' @ error/fx.c/GetOperator/2446.
convert: Expected operator at 'page.y+4' @ error/fx.c/GetOperator/2446.
convert: Expected operator at 'page.y+4' @ error/fx.c/GetOperator/2446.

I can share the specific images I used if needed, but to me the error seems like an issue with the command. Can someone verify if this command still valid for Image Magick version 7?

**Comment by fmw42** [comment]
For IM 7, use magick, not magick convert and not convert.  Try that and see if that does not error. IM 6 convert does not allow inline computations. You can only do that with IM 7 magick.

**Reply by SheriffBuzz** [reply]
I had tried that, got the same error.

![image](https://user-images.githubusercontent.com/83767022/208573233-74604bbe-f270-428b-aaa1-1aba8b3be31a.png)

![image](https://user-images.githubusercontent.com/83767022/208573425-dccb9846-e8cc-4e6c-93f5-32b3ed044263.png)

**Comment by snibgo** [comment, marked as answer]
It is successfully running fx.c/GetOperator, so it is running v7.

Try a "`.`" (dot, aka period) before the page inside each fx:
```
 '+%[fx:u[t-1].page.x+u[t-1].w-8]+%[fx:u[t-1].page.y+4]' 
```

**Reply by SheriffBuzz** [reply]
that worked. Thanks!
